src/realm/cluster.cpp

```diff
@@ -1651,32 +1660,8 @@ ref_type Cluster::typed_write(ref_type ref, _impl::ArrayWriterBase& out) const
         else {
             // Columns
             auto col_key = out.table->m_leaf_ndx2colkey[j - 1];
+            out.col_key = col_key;
             auto col_type = col_key.get_type();
-            // String columns are interned at this point
```

Comment:
I am not sure if I understood this, where is this array going to be written?

Reply:
In the general call to typed_write. (Line 1741)